finagle-redis/src/main/scala/com/twitter/finagle/redis/exp/ConnectionInitCommand.scala

```diff
@@ -0,0 +1,47 @@
+package com.twitter.finagle.redis.exp
+
+import com.twitter.finagle._
+import com.twitter.finagle.redis.ClientError
+import com.twitter.finagle.redis.param.{Database, Password}
+import com.twitter.finagle.redis.protocol._
+import com.twitter.util.Future
+
+/**
+ * A stack module that executes commands when create a new connection.
+ */
+object ConnectionInitCommand {
+
+  val Role = Stack.Role("RedisInitialCommand")
+
+  def module: Stackable[ServiceFactory[Command, Reply]] =
+    new Stack.Module2[Database, Password, ServiceFactory[Command, Reply]] {
+      val role = Role
+      val description = "Manage redis connections"
+
+      def make(db: Database, password: Password, next: ServiceFactory[Command, Reply]): ServiceFactory[Command, Reply] = {
+        for {
+          service <- next
```

Comment:
I think we need to make sure we close the `service` if any of the follow up queries fail.